### ISSUE TITLE
[EKS] [request]: Ability to add multiple oidc providers for user authentication

### ISSUE THREAD

**Issue by ugurzongur** [aws/containers-roadmap#2300]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
With the introduction of [KEP-3331 - Structured Authentication Configuration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3331-structured-authentication-configuration) it's possible to define multiple OIDC clientIDs (issuers) for user authentication. Will EKS have support for this?

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
We manage diverse entities accessing the Kubernetes API, including humans, pipeline jobs, and services that need cross-cluster access. Current authentication solutions often rely on long-lived credentials. Introducing multiple issuers would enable us to solve authentication more elegantly and securely. For instance, GitLab jobs could directly access the Kubernetes API using ID tokens, while our central service could make changes on other clusters directly using projected tokens.

**Are you currently working around this issue?**
We use OIDC for one aspect (humans) and others generally use some form of long-lived credentials (e.g. long-lived Service Account tokens).

**Additional context**
-

**Attachments**
-

**Comment by robert-heinzmann-logmein** [comment]
It seems the feature is GA now with 1.34 and EKS supports 1.34 as well, is this available ? 

<img width="775" height="154" alt="Image" src="https://github.com/user-attachments/assets/e7c7fb90-e51a-4177-9405-9c7b54857e1d" />

**Comment by cnemo-cenic** [comment]
No, this doesn't work with 1.34 at least out of the box. I have been using IRSA and registering the provider in IAM.

**Comment by danielloader** [comment]
Just watching terraform take 10 minutes to create/update/delete the OIDC identity config (which I assume is a full control plane "upgrade" and rolling of the master nodes) makes me yearn for some YAML cluster internal configuration for this.

Even if I just wanted to configure a single provider having this adopted would speed up provisioning, let alone the benefits to being able to have a couple of different IDPs for different usecases.

**Comment by mikestef9** [comment]
https://aws.amazon.com/about-aws/whats-new/2026/08/amazon-eks-multiple-oidc-providers/

**Comment by spyroang** [comment]
Amazon EKS now supports up to 10 external OIDC identity providers per cluster (Kubernetes 1.32+). To learn more, see [Grant users access to Kubernetes with an external OIDC provider ](https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html) in the Amazon EKS User Guide. Closing as delivered.

https://aws.amazon.com/about-aws/whats-new/2026/08/amazon-eks-multiple-oidc-providers/
